deploy/terraform/modules/ec2/aws-ec2.tf

```diff
@@ -0,0 +1,186 @@
+
+  load_balancers = ["${var.load_balancer_name}"]
+
+  wait_for_elb_capacity = "${var.asg_minimum_number_of_instances}"
+
+  tag {
+    key = "environment"
+    value = "${var.build_env}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key = "Name"
+    value = "datascope-backend-${var.build_env}"
+    propagate_at_launch = true
+  }
+
+  tag {
+    key = "Client"
+    value = "WBG"
+    propagate_at_launch = true
+  }
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_autoscaling_policy" "scale_up" {
+  name = "${aws_autoscaling_group.main_asg.name}-scale-up"
+  scaling_adjustment = 1
+  adjustment_type = "ChangeInCapacity"
+  cooldown = 300
+  autoscaling_group_name = "${aws_autoscaling_group.main_asg.name}"
+}
+
+resource "aws_cloudwatch_metric_alarm" "latency_high" {
+    alarm_name = "${var.load_balancer_name}-latency-high"
+    comparison_operator = "GreaterThanOrEqualToThreshold"
+    evaluation_periods = "2"
+    metric_name = "Latency"
+    namespace = "AWS/ELB"
+    period = "120"
+    statistic = "Average"
+    threshold = "0.5"
+    dimensions {
+        LoadBalancerName = "${var.load_balancer_name}"
+    }
+    alarm_description = "This metric monitors high ec2 elb latency"
+    alarm_actions = ["${aws_autoscaling_policy.scale_up.arn}"]
+}
+
+resource "aws_autoscaling_policy" "scale_down" {
+  name = "${aws_autoscaling_group.main_asg.name}-scale-down"
+  scaling_adjustment = -1
+  adjustment_type = "ChangeInCapacity"
+  cooldown = 300
+  autoscaling_group_name = "${aws_autoscaling_group.main_asg.name}"
+}
+
+resource "aws_cloudwatch_metric_alarm" "latency_low" {
+    alarm_name = "${var.load_balancer_name}-latency-low"
+    comparison_operator = "LessThanOrEqualToThreshold"
+    evaluation_periods = "2"
+    metric_name = "Latency"
+    namespace = "AWS/ELB"
+    period = "120"
+    statistic = "Average"
+    threshold = "0.25"
+    dimensions {
+        LoadBalancerName = "${var.load_balancer_name}"
+    }
+    alarm_description = "This metric monitors low ec2 elb latency"
+    alarm_actions = ["${aws_autoscaling_policy.scale_down.arn}"]
+}
+
+resource "aws_cloudwatch_metric_alarm" "cpu_utilization_high" {
+  alarm_name          = "${aws_autoscaling_group.main_asg.name}-cpu-utilization-high"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "2"
+  metric_name         = "CPUUtilization"
+  namespace           = "AWS/EC2"
+  period              = "120"
+  statistic           = "Average"
+  threshold           = "70"
+
+  dimensions {
+    AutoScalingGroupName = "${aws_autoscaling_group.main_asg.name}"
+  }
+
+  alarm_description = "This metric monitors ec2 cpu utilization"
+  alarm_actions     = ["${aws_autoscaling_policy.scale_up.arn}", "arn:aws:sns:us-west-2:844297601570:datascope", "arn:aws:sns:us-west-2:844297601570:datascope-alerts"]
+}
+
+resource "aws_security_group" "allow_all" {
+  name        = "allow_all"
+  description = "Allow all inbound traffic"
```

Comment:
change this to allow all inbound and outbound traffic

Reply:
Good catch.